control-plane/subcommand/server-acl-init/command.go

```diff
@@ -710,13 +724,32 @@ func (c *Command) Run(args []string) int {
 		// Controller token must be global because config entry writes all
 		// go to the primary datacenter. This means secondary datacenters need
 		// a token that is known by the primary datacenters.
-		err = c.createGlobalACL("controller", rules, consulDC, isPrimary, consulClient)
+		err = c.createGlobalACL("controller", rules, consulDC, isPrimary, false, consulClient)
 		if err != nil {
 			c.log.Error(err.Error())
 			return 1
 		}
-	}
 
+		policyName := "controller-token"
+		if c.flagFederation && !isPrimary {
+			// If performing ACL replication, we must ensure policy names are
+			// globally unique so we append the datacenter name but only in secondary datacenters..
+			policyName += fmt.Sprintf("-%s", consulDC)
+		}
+		ap := &api.ACLRolePolicyLink{
```

Comment:
I could not figure out how to get this to instantiate properly and pass the linter so it is hacky looking right now.